components/brave_today/browser/direct_feed_controller.cc

```diff
@@ -339,19 +340,34 @@ void DirectFeedController::OnResponse(
     std::move(callback).Run(std::move(result));
     return;
   }
-  // Reponse is valid, but still might not be a feed
-  FeedData data;
-  if (!parse_feed_string(::rust::String(body_content), data)) {
-    VLOG(1) << feed_url.spec() << " not a valid feed.";
-    VLOG(2) << "Response body was:";
-    VLOG(2) << body_content;
-    std::move(callback).Run(std::move(result));
-    return;
-  }
-  // Valid feed
-  result->success = true;
-  result->data = data;
-  std::move(callback).Run(std::move(result));
+
+  // Response is valid, but still might not be a feed
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      base::BindOnce(
+          [](const GURL& feed_url,
```

Comment:
Is this reference guaranteed to outlive the lifetime of the thread? Or should we make a copy instead?

Reply:
Good call!